packages/entities/entities-plugins/src/components/PluginCustomGrid.vue

```diff
@@ -0,0 +1,274 @@
+<template>
+  <div class="custom-plugins-grid">
+    <KEmptyState
+      v-if="!modifiedCustomPlugins.length"
+      class="custom-plugins-empty-state"
+      cta-is-hidden
+      icon="stateGruceo"
+      icon-size="96"
+    >
+      <!-- this will only be shown if not allowed to create custom plugins -->
+      <template #title>
+        <span class="empty-state-title">
+          {{ t('plugins.select.tabs.custom.empty_title') }}
+        </span>
+      </template>
+
+      <template #message>
+        <span class="empty-state-description">
+          {{ t('plugins.select.tabs.custom.empty_description') }}
+        </span>
+      </template>
+    </KEmptyState>
+
+    <KCollapse
+      v-else
+      v-model="shouldCollapsedCustomPlugins"
+      class="custom-plugins-collapse"
+      :data-testid="`${PluginGroup.CUSTOM_PLUGINS}-collapse`"
+      :title="PluginGroup.CUSTOM_PLUGINS"
+      :trigger-label="triggerLabel"
+    >
+      <!-- don't display a trigger if all plugins will already be visible -->
+      <template
+        v-if="modifiedCustomPlugins.length <= pluginsPerRow"
+        #trigger
+      >
+        &nbsp;
+      </template>
+      <template #visible-content>
+        <div class="plugin-card-container">
+          <PluginSelectCard
+            v-for="(plugin, index) in getPluginCards('visible')"
+            :key="index"
```

Comment:
nit: indexes by themselves don't make good keys; can you use a plugin id/name or something instead?

Same comment for the other v-for below